core/src/main/java/com/netflix/conductor/core/execution/tasks/Join.java

```diff
@@ -117,12 +124,17 @@ public boolean execute(
     }
 
     @Override
-    public Optional<Long> getEvaluationOffset(TaskModel taskModel, long defaultOffset) {
-        int index = taskModel.getPollCount() > 0 ? taskModel.getPollCount() - 1 : 0;
-        if (index == 0) {
+    public Optional<Long> getEvaluationOffset(TaskModel taskModel, long maxOffset) {
+        int pollCount = taskModel.getPollCount();
```

Comment:
To reduce the delay of the JOIN task when all forked tasks complete it will not be postponed until `pollCount` is above a given configurable threshold. 

Once `pollCount` is greater than that threshold, the delay will increase until reaching `maxOffset`.